locator/topology.hh

```diff
@@ -102,6 +105,59 @@ public:
 
     state get_state() const noexcept { return _state; }
 
+    bool is_joining() const noexcept {
+        switch (_state) {
+        case state::bootstrapping:
+        case state::replacing:
+            return true;
+        default:
+            return false;
+        }
+    }
+
+    bool is_normal() const noexcept {
+        return _state == state::normal;
+    }
+
+    bool is_leaving() const noexcept {
+        switch (_state) {
+        case state::being_decommissioned:
+        case state::being_removed:
+        case state::being_replaced:
+            return true;
+        default:
+            return false;
+        }
+    }
+
+    bool is_token_owner_for_read() const noexcept {
+        switch (_state) {
+        case state::normal:
```

Comment:
Currently, the rule is simple, if a node is in NORMAL status it is part of the ring. Here we reuse the term normal for a state that is a bit different.  Nothing wrong about it. But it turns out, even with the much more complicated states, is_token_owner_for_read() and is_token_owner_for_write() are not very useful, as discussed below.

Reply:
How about just leave `bool is_token_owner()` and drop the _for_read and _for_write versions?

Reply:
That"s possible.  It'd essentially be the "for read" version, while the "for write" version is a super-set of it that includes also pending nodes.

Reply:
done in v3